File: trunk/PolarMathWinRT/Evaluation/Spectrum/LF.cs
using HrmMath.Data;
using HrmMath.Util;
using System.Collections.Generic;

namespace HrmMath.Evaluation.Spectrum
{
    public sealed class LF : IEvaluation
    {
        public Index Name
        {
            get { return Index.LF; }
            set { }
        }

        public object Evaluate(SessionData training)
        {
            var periodogram = (List<Periodogram>)training.Evaluate( new Lomb() );

            return new Square( periodogram, 0.0333, 0.1 ).Calculate();
        }
    }
}

File: PolarMath/src/org/macondo/polar/evaluation/spectrum/LF.java
package org.macondo.polar.evaluation.spectrum;

import java.util.List;

import org.macondo.polar.data.Training;
import org.macondo.polar.evaluation.Evaluation;
import org.macondo.polar.util.Lomb;
import org.macondo.polar.util.Periodogram;
import org.macondo.polar.util.Square;

public class LF implements Evaluation<Double>{
	public Double evaluate(Training training) {
		List<Periodogram> periodogram = training.evaluate(new Lomb());
		
		return new Square(periodogram, 0.04, 0.15).Calculate();
	}
}
